packages/rln/src/contract/rln_contract.spec.ts

```diff
@@ -145,25 +171,43 @@ describe("RLN Contract abstraction - RLN", () => {
       formatIdCommitment(identity.IDCommitmentBigInt)
     );
 
-    const registerStub = sinon.stub().returns({
-      wait: () =>
-        Promise.resolve({
-          events: [
-            {
-              event: "MembershipRegistered",
-              args: {
-                idCommitment: formatIdCommitment(identity.IDCommitmentBigInt),
-                rateLimit: DEFAULT_RATE_LIMIT,
-                index: ethers.BigNumber.from(1)
-              }
-            }
-          ]
-        })
-    });
+    const registerStub = sinon
```

Comment:
general ask here:
`"should register a member"` test case is `150` lines long, it is not easy to follow 

and I believe some `mocks` can be made reusable